app/services/feature-flags.js

```diff
@@ -51,7 +52,11 @@ export default Service.extend({
       return obj;
     });
 
-    this.storage.setItem('travis.features', JSON.stringify(state));
+    const oldFeatureState = JSON.parse(this.storage.getItem('travis.features')) || {};
```

Comment:
Would it make sense to suppress parsing error here by using https://github.com/travis-ci/travis-web/blob/180f93652338b17fbcea4352eef48faf2c2eed6f/app/services/storage.js#L72-L78? Maybe even extract that helper somewhere to `/utils`. How does it sound to you?